### DISCUSSION TITLE
New Feature Requests

### DISCUSSION THREAD

**Discussion by rjkenefick** [enebin/Aespa#7] [Ideas]
Hello ! ,

So far this library has a ton of potential compared to other examples and also works the best lol ! Some features I would like to see to make it the complete package . There was a Library up until recently that was the go to for adding
camera functionality for a swiftUI project , but has been since archived and does not work due to various issues with event handling. Some of the features it had are below which would make this the new staple . 

- Flash 
- Tap to focus , which might be able to be done with whats already provided have not really tried yet .
- Pinch to zoom , again might be possible already with what is provided 
- double tap to flip to front facing camera 
- Lastly ability to also take normal Photos not video.

**Comment by enebin** [comment]
Hi @rjkenefick! Thank you for providing valuable feedback! Below are my thoughts on each of the opinions.

> Flash 
- Will be added on the next release #8. However, support for the flash mode is available only for the `torchMode`, where the flash is always turned on. Support for the `flashMode` will be added after the addition of the photo feature.
> Photo
- Considering it a top priority task and plan to add the feature in the near future, although it may not be immediate.
> Pinch zoom, double tap, tap to focus
- I initially thought any tasks related to the UI were not within the scope of this package. I believed that these tasks could be implemented by users themselves using the methods provided by our package.  However, I don't have a negative opinion about it. Perhaps in the near future (although not immediately), I will consider adding support for these tasks. If that happens, it is likely to be specifically for SwiftUI.

I am currently focusing on testing and improving the interface to create a sustainable environment. While some simple features will be added soon, please be aware that their implementation may be delayed. I always look forward to your feedback. Have a great day, my friend!

**Comment by enebin** [comment]
Hi @rjkenefick! I'd like to let you know that the new version, featuring the photo capturing capability, is now available. 

The other features you mentioned, integration of features for the SwiftUI preview instance that is currently used, is planned for the next release. Stay tuned for updates. Thanks!
